### DISCUSSION TITLE
cleanStores does not clean stores as expected

### DISCUSSION THREAD

**Discussion by kasir-barati** [ai/nanostores#168] [General]
Hi dear readers

I used nanostores in a small app of mine. Problem, I used to believe that when I write
```ts
afterEach(async () => {
    cleanStores(states.bingo, states.facedUpCards);
});
```
it will clean those caches for me but not really. here you can see and try it on your own: https://github.com/kasir-barati/bingo-game

You can see the test cases here: https://github.com/kasir-barati/bingo-game/blob/main/src/FacedUpCards.store.spec.ts
And the stores here: https://github.com/kasir-barati/bingo-game/blob/main/src/FacedUpCards.store.ts

Any idea what's wrong?

**Comment by ai** [comment]
`cleanStores` means clean stores from listeners (and trigger `onStop`). It will not reset it to initial values (because we do not store initial value anywhere for simplicity). In your case, it is better to do reset manually in the same `afterEach`.

**Comment by kasir-barati** [comment]
And what do you mean by doing it manually @ai ? I am really confused. Would you mind if I ask you to share a sample code on how I can do that?

**Comment by ai** [comment]
```ts
afterEach(async () => {
    cleanStores(states.bingo, states.facedUpCards);
    states.facedUpCards.set([])
    states.bingo.set(undefined)
});
```

**Comment by kasir-barati** [comment]
Thanks @ai but unfortunately it is impossible. I mean hypothetically, because I said that `bingo: ReadableAtom<boolean | undefined>;` not WriteableAtom to prevent developers from touching the store (the same with `facedUpCards`).

Now can we say that it was a wrong decision?

**Comment by ai** [comment]
You can use `// @ts-expect-error` to disable types here. For me, it is OK to do it for mocks in tests, since they are always goes against the public API.

**Comment by kasir-barati** [comment]
Hmmm, Really great. I would love to add our conversation as part of the README.md or documentation. To be honest I'll write a short article in my LinkedIn about it but official doc is something that everybody look at it. Just a suggestion.

Thanks a lot :heart:
